src/iot/commands/detachPolicy.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as localizedText from '../../shared/localizedText'
+import { getLogger } from '../../shared/logger'
+import { addCodiconToString } from '../../shared/utilities/textUtilities'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { Commands } from '../../shared/vscode/commands'
+import { Window } from '../../shared/vscode/window'
+import { IotCertWithPoliciesNode } from '../explorer/iotCertificateNode'
+import { showViewLogsMessage, showConfirmationMessage } from '../../shared/utilities/messages'
+import { IotPolicyNode } from '../explorer/iotPolicyNode'
+import { IotPolicyFolderNode } from '../explorer/iotPolicyFolderNode'
+
+const DELETE_FILE_DISPLAY_TIMEOUT_MS = 2000
+
+/**
+ * Detaches an IoT Policy from a certificate.
+ *
+ * Prompts the user for confirmation.
+ * Detaches the policy.
+ * Refreshes the parent node.
+ */
+export async function detachPolicyCommand(
+    node: IotPolicyNode,
+    window = Window.vscode(),
+    commands = Commands.vscode()
+): Promise<void> {
+    getLogger().debug('DetachPolicy called for %O', node)
+
+    const policyName = node.policy.name
+    if (node.parent instanceof IotPolicyFolderNode) {
+        return undefined
+    }
```

Comment:
Should the command even be available in this case?

Reply:
I'm going to make this node into an abstract class and extend it into two separate classes depending on the type of the parent node, so that we don't have to have code like this. I'll do that in another PR though